executor/join.go

```diff
@@ -277,6 +290,28 @@ func (e *HashJoinExec) fetchInnerRows(ctx context.Context, chkCh chan<- *chunk.C
 	}
 }
 
+// evalRadixBitNum evaluates the radix bit numbers.
+func (e *HashJoinExec) evalRadixBitNum() {
+	sc := e.ctx.GetSessionVars()
+	// Calculate the bit number needed when using radix partition.
+	if sc.EnableRadixJoin {
+		return
+	}
+	innerResultSize := float64(e.innerResult.GetMemTracker().BytesConsumed())
+	// To ensure that one partition of inner relation, one hash
+	// table and one partition of outer relation fit into the L2
+	// cache when the input data obeys the uniform distribution,
+	// we suppose every sub-partition of inner relation using
+	// three quarters of the L2 cache size.
+	l2CacheSize := float64(sc.L2CacheSize) * 3 / 4
+	e.radixBits = int(math.Log2(innerResultSize / l2CacheSize))
+	// The complete inner relation can be hold in L2Cache, we
+	// can skip the partition phase.
+	if e.radixBits <= 0 {
+		sc.EnableRadixJoin = false
```

Comment:
If a query in a session set `EnableRadixJoin` to `false`, all subsequent queries cannot use radix hash join? shall we put it in `StatementContext`?

Reply:
@eurekaka 
I think it seems to be ok.

Now we will not use radix hash join in two cases:
1. The full size of the inner table is < the L2 cache size.
2. The data of the inner table is high skewed.

Both of the 2 cases are detected during executing,
we do not really need a variable in StatementContext to check whether to use radix hash join for every HashJoinExec.

Reply:
Hmm, for example:
```
-- t1 10 rows, t2 10000 rows, t3 10 rows
-- this query would set sc. EnableRadixJoin to false since t2 is larger than L2 cache size
select * from t1 left join t2 on t1.c1 = t2.c1;
-- sc.EnableRadixJoin has been set false, we do not use radix hash join for this query?
select * from t1 left join t3 on t1.c1 = t3.c1;
```

Reply:
What I mean here was that:
EnableRadixJoin is a variable used before the algorithm is fully implemented,
since we need it to do testing manually.
We should check whether to use radix hash join dynamically in the end.

Reply:
OK got it. LGTM.

Reply:
I removed this check. Since it's useless now. @eurekaka